telegrambots-client/src/main/java/org/telegram/telegrambots/client/okhttp/AbstractOkHttpFutureCallback.java

```diff
@@ -0,0 +1,54 @@
+package org.telegram.telegrambots.client.okhttp;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiRequestException;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+abstract class AbstractOkHttpFutureCallback<T> extends CompletableFuture<T> implements Callback {
+    @Override
+    public void onFailure(@NonNull Call call, @NonNull IOException exception) {
+        completeExceptionally(exception);
+    }
+
+    @Override
+    public void onResponse(@NonNull Call call, @NonNull Response response) {
+        if (!response.isSuccessful()) {
+            var messageBuilder = new StringBuilder();
+            messageBuilder.append("code: ").append(response.code());
+            messageBuilder.append(", message: ").append(response.message());
+
+            if(response.body() != null) {
```

Comment:
are you sure you can do it? shouldn't response.body() be called only once? multiple calls will cause an exception

Reply:
and all future of past calls too

Reply:
It just returns a value of the `body` field
This `if` just a check that body exists and accessing it will not give us an NPE

But the `body` is a `ResponseBody` instance and is consumable and should be consumed once

And here I call `.string()` which consumes body and closes it, so this piece does not require `try-with-resources` notation